src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -136,17 +127,70 @@ void CPlayer::drawDeathEffect(HexagonGame& mHexagonGame)
     }
 }
 
-void CPlayer::playerSwap(HexagonGame& mHexagonGame, bool mPlaySound)
-{
+void CPlayer::playerSwap(HexagonGame &mHexagonGame, bool mPlaySound) {
     angle += ssvu::pi;
     mHexagonGame.runLuaFunctionIfExists<void>("onCursorSwap"); // TODO: docs
 
-    if(mPlaySound)
-    {
+    if (mPlaySound) {
         mHexagonGame.getAssets().playSound("swap.ogg");
     }
 }
 
+[[nodiscard]] sf::Vector2f CPlayer::getPosition() const {
+    return pos;
+}
+
+[[nodiscard]] sf::Vector2f CPlayer::getStartPosition() const {
+    return startPos;
+}
+
+void CPlayer::kill(HexagonGame &mHexagonGame, bool mEffect) {
+    deadEffectTimer.restart();
+
+    if (!Config::getInvincible()) { dead = true; }
+    if (mEffect) { mHexagonGame.death(); }
+
+    ssvs::moveTowards(lastPos, ssvs::zeroVec2f, 5 * mHexagonGame.getSpeedMultDM());
+
+    pos = lastPos;
+}
+
+void CPlayer::push(HexagonGame& mHexagonGame, CWall& wall){
+    const auto curveData = wall.getCurve();
+    const int curveDir = ssvu::getSign(curveData.speed);
+    const int movement{mHexagonGame.getInputMovement()};
+
+    const float radius{mHexagonGame.getRadius()};
+
+    unsigned int maxAttempts;
+    if (curveDir != 0){
+        maxAttempts = std::abs(curveData.speed+speed*(movement*curveDir));
+    } else {
+        maxAttempts = speed;
+    }
```

Comment:
```suggestion
    const unsigned int maxAttempts = curveDir != 0 
        ? std::abs(curveData.speed+speed*(movement*curveDir)) : speed;
```

Reply:
This is good because it allows us to mark `maxAttempts` as `const`, so we know it won't change in the rest of the function